lib/logger.js

```diff
@@ -13,17 +13,34 @@ var DEF_LEVEL_INDEX = LEVELS.indexOf(DEF_LEVEL);
 
 // Simple bunyan logger wrapper
 function Logger(confOrLogger, args) {
+    var self = this;
     if (confOrLogger.constructor !== Logger) {
         // Create a new root logger
         var conf = this._processConf(confOrLogger);
-        this._logger = bunyan.createLogger(conf);
-        this._levelMatcher = this._levelToMatcher(conf.level);
+        self._log_components = conf.log_components || {};
+        delete conf.log_components;
+        self._logger = bunyan.createLogger(conf);
+        self._levelMatcher = this._levelToMatcher(conf.level);
+
+        // For each specially logged component we need to create
+        // a child logger that accepts everything regardless of the level
+        self._componentLoggers = {};
+        Object.keys(self._log_components).forEach(function(component) {
+            self._log_components[component] =
+                Math.ceil(self._log_components[component] * 1024);
```

Comment:
@gwicke According to @d00rman we will have better precision with this. The log probabilities could be extremely low with high request rates. But now I'm starting to thing this is not actually needed, since we're just comparing, all floating point issues would not be applicable here. Will remove.